stock_secondary_unit/models/stock_move.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2018 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+from odoo.tools.float_utils import float_round
+
+
+class StockMove(models.Model):
+    _inherit = ["stock.move", "product.secondary.unit.mixin"]
+    _name = "stock.move"
+    _secondary_unit_fields = {
+        "qty_field": "product_uom_qty",
+        "uom_field": "product_uom",
+    }
+
+    product_uom_qty = fields.Float(
+        store=True,
+        readonly=False,
+        compute="_compute_product_uom_qty",
+        copy=True,
+        precompute=True,
+    )
+
+    @api.depends("secondary_uom_qty", "secondary_uom_id")
+    def _compute_product_uom_qty(self):
+        self._compute_helper_target_field_qty()
+
+    @api.onchange("product_uom")
+    def onchange_product_uom_for_secondary(self):
+        self._onchange_helper_product_uom_for_secondary()
+
+    @api.model
+    def _prepare_merge_moves_distinct_fields(self):
+        """Don't merge moves with distinct secondary units"""
+        distinct_fields = super()._prepare_merge_moves_distinct_fields()
+        distinct_fields += ["secondary_uom_id"]
+        return distinct_fields
+
+
+class StockMoveLine(models.Model):
+    _inherit = ["stock.move.line", "product.secondary.unit.mixin"]
+    _name = "stock.move.line"
+    _secondary_unit_fields = {"qty_field": "qty_done", "uom_field": "product_uom_id"}
+
+    qty_done = fields.Float(
+        store=True, readonly=False, compute="_compute_qty_done", precompute=True
+    )
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            move = self.env["stock.move"].browse(vals.get("move_id", False))
+            if move.secondary_uom_id:
+                uom = self.env["uom.uom"].browse(vals["product_uom_id"])
+                factor = move.secondary_uom_id.factor * uom.factor
+                move_line_qty = vals.get("product_uom_qty", vals.get("qty_done", 0.0))
```

Comment:
I am trying to wrap my head around this because whatever I do the lines end up with 0 secondary qty and None secondary uom id. Why? Because move_secondary_uom_id is None. I cannot find any piece of code that would ever be attempting to write any value there. And I have looked and tested versions 15 and 14 as well. Any insights on this?

Reply:
So when I confirm Purchase Order and have a look at associated stock picking this is what I get always. Or am I expecting wrongly to have secondary qty and uom displayed here?
![image](https://github.com/OCA/stock-logistics-warehouse/assets/19169467/0eb7e836-2e6d-47a9-aa10-d63df77a42ac)

Reply:
In my opinion stock.move is missing something like this:
```
    @api.model_create_multi
    def create(self, vals_list):
        for vals in vals_list:
            product = self.env["product.product"].browse(vals.get("product_id", False))
            if product:
                vals.update(
                    {
                        "secondary_uom_id": product.stock_secondary_uom_id.id
                        or product.product_tmpl_id.stock_secondary_uom_id.id,
                    }
                )
        return super().create(vals_list)

```

After that the result is what (at least) I expect:
![image](https://github.com/OCA/stock-logistics-warehouse/assets/19169467/5d4181cb-1016-4ae7-8582-d23da1837fb8)